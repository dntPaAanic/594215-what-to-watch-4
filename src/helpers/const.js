const PREVIEW_DELAY = 1000;

const SIMILAR_FILMS_COUNT = 4;

const RatingLevel = {
  BAD: 0,
  NORMAL: 3,
  GOOD: 5,
  VERY_GOOD: 8,
  AWESOME: 10
};

export {PREVIEW_DELAY, SIMILAR_FILMS_COUNT, RatingLevel};
