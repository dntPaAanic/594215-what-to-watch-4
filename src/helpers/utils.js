const PREVIEW_DELAY = 1000;
const RatingLevel = {
  BAD: 0,
  NORMAL: 3,
  GOOD: 5,
  VERY_GOOD: 8,
  AWESOME: 10
};

const getRatingLevel = (ratingScore) => {
  switch (true) {
    case (ratingScore < RatingLevel.NORMAL):
      return `Bad`;
    case (ratingScore < RatingLevel.GOOD):
      return `Normal`;
    case (ratingScore < RatingLevel.VERY_GOOD):
      return `Good`;
    case (ratingScore < RatingLevel.AWESOME):
      return `Very good`;
    case (ratingScore === RatingLevel.AWESOME):
      return `Awesome`;
    default:
      break;
  }
  return null;
};

export {PREVIEW_DELAY, getRatingLevel};
