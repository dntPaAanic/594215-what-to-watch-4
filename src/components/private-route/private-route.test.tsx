import * as React from 'react';
import * as renderer from 'react-test-renderer';
import {Provider} from 'react-redux';
import {Router} from 'react-router-dom';
import history from '../../history';
import configureStore from 'redux-mock-store';
import PrivateRoute from './private-route';
import Namespace from '../../reducer/name-space';
import {films} from '../../helpers/test-data';

it(`PrivateRoute should render correct`, () => {
  const mockStore = configureStore([]);

  const store = mockStore({
    [Namespace.FILMS]: {
      movieCards: films,
      mainMovie: films[0],
      filterType: `All genres`,
      showingCards: 8,
      isAppLoading: false
    },
    [Namespace.USER]: {
      authorizationStatus: false,
    },
    [Namespace.COMMENTS]: {
      comments: [],
      isCommentsLoaded: false
    }
  });

  const tree = renderer.create(
      <Router history={history}>
        <Provider store={store}>
          <PrivateRoute exact path={`/login`} isAuthed={false} render={() => HTMLElement as React.ReactNode}/>
        </Provider>
      </Router>, {
        createNodeMock: () => {
          return {};
        }}).toJSON();

  expect(tree).toMatchSnapshot();
});
