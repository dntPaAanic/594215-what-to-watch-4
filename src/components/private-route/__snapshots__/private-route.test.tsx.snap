// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrivateRoute should render correct 1`] = `null`;
